--- compose.inc.orig	2026-05-14 17:53:34.936531010 +0200
+++ compose.inc	2026-05-14 17:54:40.943013776 +0200
@@ -246,6 +246,9 @@
             $dns = $db->get('conf.system.network.dns');
             $urlhn = $dns->get('hostname');
             $dn = $dns->get('domainname');
+            if (empty($dn)) { 
+                $dn = "local";
+            }
             if (strlen($dn) > 1) {
                 $urlhn = sprintf('%s.%s', $urlhn, $dn);
             }
