--- compose.inc.orig	2026-04-24 20:41:52.368897757 +0200
+++ compose.inc	2026-04-24 20:43:01.942499093 +0200
@@ -242,6 +242,9 @@
             $dns = $db->get('conf.system.network.dns');
             $urlhn = $dns->get('hostname');
             $dn = $dns->get('domainname');
+            if (empty($dn)) { 
+                $dn = "local";
+            }
             if (strlen($dn) > 1) {
                 $urlhn = sprintf('%s.%s', $urlhn, $dn);
             }
