--- default.sls.orig	2026-05-14 18:47:54.317706770 +0200
+++ default.sls	2026-05-14 18:49:27.770426832 +0200
@@ -24,5 +24,5 @@
 
 configure_hostname:
   cmd.run:
-    - name: hostnamectl set-hostname "{{ config.hostname | lower }}"
-    - unless: test "{{ config.hostname | lower }}" = "$(hostname)"
+    - name: hostnamectl set-hostname "{{ config.hostname }}"
+    - unless: test "{{ config.hostname }}" = "$(hostname)"
