--- mdmgmt.inc.orig	2026-06-03 18:29:46.039593236 +0200
+++ mdmgmt.inc	2026-06-03 18:33:21.522795954 +0200
@@ -253,10 +253,6 @@
             if (true === $sd->isReadOnly()) {
                 continue;
             }
-            // Skip devices that are connected via USB.
-            if (true === $sd->isUsb()) {
-                continue;
-            }
             // Check if the device is referenced/used by a plugin.
             $db = \OMV\Config\Database::getInstance();
             if (true === $db->exists("conf.service", [
