--- sharemgmt.inc.orig	2026-04-24 20:29:35.974654406 +0200
+++ sharemgmt.inc	2026-04-24 20:34:39.358890851 +0200
@@ -1127,6 +1127,8 @@
 						$entryv['name'], $entryv['perms']);
 				}
 			}
+			// only set default acl if named users / groups exist
+			if (!empty($aclSpec)) {
 			// Set default owner permissions.
 			if (array_key_exists("userperms", $params)) {
 				$aclSpec[] = sprintf("default:user::%d", $params['userperms']);
@@ -1159,6 +1161,13 @@
 				$cmdArgs[] = sprintf("-M %s", escapeshellarg(
 					$aclSpecFile->getFileName()));
 			}
+			} else {
+				// delete acl if no named users / groups
+				$cmdArgs = ["--remove-all"];
+				if (TRUE === boolvalEx($params['recursive'])) {
+					$cmdArgs[] = "--recursive";
+				}
+			}
 			$cmdArgs[] = "--";
 			$cmdArgs[] = escapeshellarg(build_path(DIRECTORY_SEPARATOR,
 				$sfpath, $params['file']));
